ui/appui-react/src/appui-react/widgets/useThisWidget.tsx

```diff
@@ -0,0 +1,102 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import React from "react";
+import { getTabLocation, TabIdContext } from "@itwin/appui-layout-react";
+import { assert } from "@itwin/core-bentley";
+import { useActiveFrontstageDef } from "../frontstage/FrontstageDef";
+import { InternalFrontstageManager } from "../frontstage/InternalFrontstageManager";
+import { UiItemsManager } from "../ui-items-provider/UiItemsManager";
+import type { WidgetState } from "./WidgetState";
+
+/** Interface to be used with the useThisWidget hook. Holds information
+ * about the Widget such as the state and location. Also contains a function to set WidgetState.
+ * @alpha
+ */
+export interface ThisWidget {
+  widgetLocation: "docked" | "floating" | "popout";
+  setState(state: Exclude<WidgetState, WidgetState.Floating>): void;
+  state: WidgetState;
+}
+
+/** Hook that returns information about the Widget in the current context.
+ * @returns ThisWidget interface that contains the WidgetLocation, WidgetState, and
+ * the ability to set the state of the widget.
+ * @alpha
+ */
+export function useThisWidget(): ThisWidget {
```

Comment:
Will we need a hook to get information about other widgets? Should we instead consider something like:
`useWidget(widgetId? string): Widget;`, for now at least we could just go with `useWidget()`, but would probably need to come up with a better name for `ThisWidget`.

Reply:
I'd rather not suggest giving access to other widget with this one, the goal being that this will be a tight link to yourself, if we add access to other widget, then the potential state solution we were thinking will be really awkward as any widget will be able to have an impact on any other widgets, hence the suggestion for "ThisWidget" (which we can change, but that was the idea). If needed, we can have a different hook which may give a different set of tools to other widget.

Reply:
Ill rename it to useWidget()

Reply:
removed interface, renamed it to useWidget()